packages/sync-service/lib/electric/config.ex

```diff
@@ -374,4 +374,23 @@ defmodule Electric.Config do
       {:error, message} -> raise Dotenvy.Error, message: message
     end
   end
+
+  def validate_security_config!(secret, insecure) do
+    cond do
+      insecure && secret != nil ->
+        raise "You cannot set both ELECTRIC_SECRET and ELECTRIC_INSECURE=true"
+
+      !insecure && secret == nil ->
+        raise "You must set ELECTRIC_SECRET unless ELECTRIC_INSECURE=true. Setting ELECTRIC_INSECURE=true risks exposing your database, only use insecure mode in development or you've otherwise secured the Electric API"
```

Comment:
Another small copy tweak:

```suggestion
        raise "You must set ELECTRIC_SECRET or set ELECTRIC_INSECURE=true. Setting ELECTRIC_INSECURE=true risks exposing your database, only use insecure mode in development or you've otherwise secured the Electric API"
```